workers/unity/Packages/io.improbable.gdk.testutils/MockCommandSender.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Unity.Entities;
+
+namespace Improbable.Gdk.TestUtils
+{
+    public class MockCommandSender : ICoreCommandSender
+    {
+        private readonly MockWorld world;
+
+        private long nextRequestId = 1;
+
+        private readonly Dictionary<Type, SortedSet<long>> requestIds = new Dictionary<Type, SortedSet<long>>();
+
+        private readonly Dictionary<long, ICommandRequest> outboundRequests = new Dictionary<long, ICommandRequest>();
+
+        internal MockCommandSender(MockWorld world)
+        {
+            this.world = world;
+        }
+
+        public IEnumerable<(CommandRequestId, TRequest)> GetOutboundCommandRequests<TRequest>() where TRequest : ICommandRequest
+        {
+            return requestIds[typeof(TRequest)].Select(id => (new CommandRequestId(id), (TRequest) outboundRequests[id]));
+        }
```

Comment:
🤔 Do you guys think this is needed?

Reply:
So this is all requests that haven't had a response yet? I think it could be useful 🤷‍♂️ 

but I'm trying to think a test case that would want it right now

Reply:
It may be that `IEnumerable<CommandRequestId>` would be sufficient for the use case of 'all requests that haven't had a response yet'. We can also choose to omit this and add it later.